File: code_samples/csharp/utilities_regex.cs
using System;

/// <summary>
/// Utilities: Regex
/// AI/ML Training Sample
/// </summary>
public class Regex
{
    private string data;
    
    public Regex()
    {
        this.data = string.Empty;
    }
    
    public void Process(string input)
    {
        this.data = input;
    }
    
    public string GetData()
    {
        return this.data;
    }
    
    public bool Validate()
    {
        return !string.IsNullOrEmpty(this.data);
    }
    
    public static void Main(string[] args)
    {
        var instance = new Regex();
        instance.Process("example");
        Console.WriteLine($"Data: {instance.GetData()}");
        Console.WriteLine($"Valid: {instance.Validate()}");
    }
}

File: code_samples/go/utilities_regex.go
package main

import (
    "fmt"
)

// Utilities: Regex
// AI/ML Training Sample

type Regex struct {
    Data string
}

func NewRegex() *Regex {
    return &Regex{
        Data: "",
    }
}

func (s *Regex) Process(input string) {
    s.Data = input
}

func (s *Regex) Validate() bool {
    return len(s.Data) > 0
}

func (s *Regex) GetData() string {
    return s.Data
}

func main() {
    instance := NewRegex()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
